slow_query_exporter.rb

```diff
@@ -0,0 +1,230 @@
+    skip:     /^(# Time: \d+|use \w+;)/,
+    userhost: /^# User@Host: (\S+)\[\S+\] @\s+\[(\S+)\]/,
+    thread:   /^# Thread_id: (\d+)\s+Schema: (\w+)\s+Last_errno: (\d+)\s+Killed: (\d+)/,
+    qtime:    /^# Query_time: (\S+)\s+Lock_time: (\S+)\s+Rows_sent: (\d+)\s+Rows_examined: (\d+)\s+Rows_affected: (\d+)\s+Rows_read: (\d+)/,
+    bytes:    /^# Bytes_sent: (\d+)\s+Tmp_tables: (\d+)\s+Tmp_disk_tables: (\d+)\s+Tmp_table_sizes: (\d+)/,
+    trxid:    /^# InnoDB_trx_id: (\S+)/,
+    qchit:    /^# QC_Hit: (\w+)\s+Full_scan: (\w+)\s+Full_join: (\w+)/,
+    filesort: /^# Filesort: (\w+)\s+Filesort_on_disk: (\w+)\s+Merge_passes: (\d+)/,
+    innoio:   /^#\s+InnoDB_IO_r_ops: (\d+)\s+InnoDB_IO_r_bytes: (\d+)\s+InnoDB_IO_r_wait: (\S+)/,
+    innowait: /^#\s+InnoDB_rec_lock_wait: (\S+)\s+InnoDB_queue_wait: (\S+)/,
+    innopage: /^#\s+InnoDB_pages_distinct: (\d+)/,
+    time:     /^SET timestamp=(\d+)/,
+    query:    /^(SELECT|INSERT|UPDATE|DELETE)\b/,
+  }
+
+  attr_accessor :attributes
+
+  def initialize
+    @attributes = {query_string: ""}
+    @done = false
+  end
+
+  def parse_line(line)
+    return if line =~ PATTERNS[:header] || line =~ PATTERNS[:skip]
+
+    case line
+    when PATTERNS[:userhost]
+      attributes[:user] = $1
+      attributes[:host] = $2
+    when PATTERNS[:thread]
+      attributes[:thread_id] = $1.to_i
+      attributes[:schema] = $2
+      attributes[:errno] = $3.to_i
+      attributes[:killed] = $4.to_i > 0
+    when PATTERNS[:qtime]
+      attributes[:request_time] = $1.to_f
+      attributes[:lock_time] = $2.to_f
+      attributes[:rows_sent] = $3.to_i
+      attributes[:rows_examined] = $4.to_i
+      attributes[:rows_affected] = $5.to_i
+      attributes[:rows_read] = $6.to_i
+    when PATTERNS[:bytes]
+      attributes[:bytes_sent] = $1.to_i
+      attributes[:tmp_tables] = $2.to_i
+      attributes[:tmp_disk_tables] = $3.to_i
+      attributes[:tmp_table_sizes] = $4.to_i
+    when PATTERNS[:trxid]
+      attributes[:transaction_id] = $1
+    when PATTERNS[:qchit]
+      attributes[:used_query_cache] = $1 == "Yes"
+      attributes[:full_scan] = $2 == "Yes"
+      attributes[:full_join] = $3 == "Yes"
+    when PATTERNS[:filesort]
+      attributes[:filesort] = $1 == "Yes"
+      attributes[:filesort_on_disk] = $2 == "Yes"
+      attributes[:merge_passes] = $3.to_i
+    when PATTERNS[:innoio]
+      attributes[:io_read_ops] = $1.to_i
+      attributes[:io_read_bytes] = $2.to_i
+      attributes[:io_read_wait] = $3.to_f
+    when PATTERNS[:innowait]
+      attributes[:io_lock_wait] = $1.to_f
+      attributes[:io_queue_wait] = $2.to_f
+    when PATTERNS[:innopage]
+      attributes[:io_distinct_pages] = $1.to_i
+    when PATTERNS[:time]
+      attributes[:timestamp] = $1.to_i
+    when PATTERNS[:query]
+      append_line(line)
+    else
+      if attributes[:query_string]
+        append_line(line)
+      else
+        raise "Weird-looking log line:\n#{line}\n"
```

Comment:
I'm not sure who will be reviewing this output - but is it possible this message might be unclear to them?

Maybe 
```
Do not know how to parse log line:
```
Or something? (unless you think it's clear enough for anyone viewing Greylog)